esphome/components/gpio/switch/__init__.py

```diff
@@ -13,6 +13,8 @@
     "RESTORE_DEFAULT_ON": GPIOSwitchRestoreMode.GPIO_SWITCH_RESTORE_DEFAULT_ON,
     "ALWAYS_OFF": GPIOSwitchRestoreMode.GPIO_SWITCH_ALWAYS_OFF,
     "ALWAYS_ON": GPIOSwitchRestoreMode.GPIO_SWITCH_ALWAYS_ON,
+    "RESTORE_INVERTED_DEFAULT_OFF": GPIOSwitchRestoreMode.GPIO_SWITCH_RESTORE_INVERTED_DEFAULT_OFF,
```

Comment:
```suggestion
    "RESTORE_TOGGLE_DEFAULT_OFF": GPIOSwitchRestoreMode.GPIO_SWITCH_RESTORE_TOGGLE_DEFAULT_OFF,
```

Just a minor thing, but `inverted` sounds quite cryptic, where `toggle` explains the intended effect quite well (downside is that it's a verb, not an adjective)

Reply:
What about `toggled` ?

Reply:
I think "toggle" is an action; that doesn't cover it.
I originally wrote that first and quickly disliked it. What does it toggle?

The fact is, it restores and inverts the value :)

Reply:
Guys got this figured out already?

Please be concrete on the naming that needs to be changed.